engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/compile/Validations.scala

```diff
@@ -1,24 +1,67 @@
 package pl.touk.nussknacker.engine.compile
 
 import cats.data.Validated.{invalid, valid}
-import cats.data.ValidatedNel
+import cats.data._
 import cats.syntax.apply._
-import pl.touk.nussknacker.engine.api.context.ProcessCompilationError.{MissingParameters, NodeId, RedundantParameters}
-import pl.touk.nussknacker.engine.api.context.{PartSubGraphCompilationError, ProcessCompilationError}
+import pl.touk.nussknacker.engine.api.context.ProcessCompilationError.{MissingParameters, NodeId, ParameterValidationError, RedundantParameters}
+import pl.touk.nussknacker.engine.api.context._
+import pl.touk.nussknacker.engine.api.definition.{Parameter, ParameterValidator}
+import pl.touk.nussknacker.engine.graph.evaluatedparam
 
 object Validations {
 
-  def validateParameters[T >: PartSubGraphCompilationError <: ProcessCompilationError](definedParamNamesSet: Set[String],
-                                                                                       usedParamNamesSet: Set[String])
+  def validateParameters[T >: PartSubGraphCompilationError <: ProcessCompilationError](parameterDefinitions: List[Parameter],
+                                                                                       parameters: List[evaluatedparam.Parameter])
                                                                                       (implicit nodeId: NodeId): ValidatedNel[T, Unit] = {
+    val definedParamNamesSet = parameterDefinitions.map(_.name).toSet
+    val usedParamNamesSet = parameters.map(_.name).toSet
+
+    val validatedRedundant = validateRedundancy(definedParamNamesSet, usedParamNamesSet)
+    val validatedMissing = validateMissingness(definedParamNamesSet, usedParamNamesSet)
+    val validatedCustom = validateWithCustomValidators(parameterDefinitions, parameters)
```

Comment:
This is not the best place to invoke this validators. The best option will be somewhere after `ExpressionEvaluator.evaluate` and/or `ExpressionEvaluator.evaluateParameters`. and real check in validator should be `if (evaluatedValue == null) then invalid(...)`. Consider validator `GreaterThan(5)` and expression `2 + 4`. In this place you won't be able to validate it. But ... I don't think that it is a blocker for this PR (we can fix it in some next PR). But please think about it a moment - maybe you will find good place to move this code into.

Reply:
Ok, will fix it in next PR, cause it's not obvious where to put these validations

Reply:
Can you leave here a TODO that this validations should be in other place (check evaluated result of expression)?

Reply:
Done :)